### PR TITLE
fix: Remove expensive logs table migration

### PR DESCRIPTION
**Pull request by etr2460**
### SUMMARY
Addresses issues with https://github.com/apache/incubator-superset/pull/11714 that prevented the migration from being run on MySQL dbs with large `logs` tables.

Does the following:
- Updates the original migration to remove the add column statements
- Adds a new migration to remove the columns if they exist
- Pushes the new columns into the json column in the logs table

Does not:
- Address any naming concerns about the new columns

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2020-12-03 at 4 45 49 PM

### TEST PLAN
- run the original migration, then upgrade past my new migration without errors
- run the updated original migration, then upgrade past my new migration without errors
- run the original migration, then downgrade without errors
- run the update original migration, then downgrade without errors
- run a query and see the new fields in the json column of the logs table

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [x] Requires DB Migration.
- [x] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

to: @mistercrunch @john-bodley @dpgaspar @ktmud @graceguo-supercat

### CODE REVIEW

File: superset/utils/log.py
Comment (ktmud) [inline superset/utils/log.py:229]: This should probably happen in `log_context` so you don't have to pass these arguments around.

Comment (etr2460) [reply superset/utils/log.py]: done

File: superset/migrations/shared/utils.py
Comment (john-bodley) [inline superset/migrations/shared/utils.py:39]: Maybe, 

```python
return any(col["name"] == column for col in nsp.get_columns(table))
```

Comment (etr2460) [reply superset/migrations/shared/utils.py:39]: fyi: https://stackoverflow.com/a/52865284

but i'm realizing now, this is a pretty janky function. let me fix it

File: superset/migrations/versions/811494c0cc23_remove_path_from_logs.py
Comment (john-bodley) [inline superset/migrations/versions/811494c0cc23_remove_path_from_logs.py:38]: @etr2460 this isn't quite right. If the column exists one should really iterate over the non-NULL values and migrate those to the `json` column. That could be somewhat of an expensive operation but given the relevancy (or lack there of) and recency of the original migration it may be ok to accept the data loss.

Comment (etr2460) [reply superset/migrations/versions/811494c0cc23_remove_path_from_logs.py:38]: yeah, i made the executive decision to drop any extra metadata that came in in the last week from the previous change. I think it's reasonable since i doubt many folks are relying on them thus far

Comment (mistercrunch) [reply superset/migrations/versions/811494c0cc23_remove_path_from_logs.py:38]: I think it's ok to drop the metadata

**Review by mistercrunch — APPROVED** [review]
LGTM, thank you for picking up the pieces here.

For the record, listing out some reasons why we don't recommend using the `DBEventLogger` at scale:
- analytics event logging happens in the main thread and adds a tax to every logged request (slows down the web server a bit)
- puts significant extra load on the metadata database, that's already busy with the OLTP workload
- `logs` quickly becomes the largest table in your database [by far!], making backup and restore operations much more expensive and time consuming than they should be

For the record, at [Preset](https://preset.io) we use [Segment](https://segment.com/) and configure it to send our logs to BigQuery.

**Comment by ktmud** [comment]
I think it's useful to keep logs for things like object creation/deletion for auditing purposes. But for UX analytics and simple access logs, we should probably target moving it out of the metadata database?

Should we create two separate loggers for this?

File: superset/utils/log.py
Comment (ktmud) [inline superset/utils/log.py:95]: can we name this to `path_no_param` in case we want to update the logic in `strip_int_from_path` as discussed in #11714 ?

**Review by ktmud — APPROVED** [review]
LGTM with one naming suggestion.

File: superset/utils/log.py
Comment (etr2460) [reply superset/utils/log.py:95]: as mentioned in the summary, this PR:
> Does not:
>
> - Address any naming concerns about the new columns

I'd rather keep this PR just scoped to fixing the migration issue. We can fast follow with naming changes if we all agree they're preferred.

Comment (ktmud) [reply superset/utils/log.py:95]: I think `path_no_param` was agreed on. Just thought it's better to make that change sooner so we don't have any dirty data.

Comment (ktmud) [reply superset/utils/log.py:95]: But it you just merge this after CI is agree, I can make another follow PR soon.

Comment (mistercrunch) [reply superset/utils/log.py:95]: Feel free to completely drop `path_no_int` and the `strip_int_from_path` method

Comment (mistercrunch) [reply superset/utils/log.py:95]: I'm also ok fine with `ref` -> `object_ref` too. Happy to do in a fast follow PR

Comment (john-bodley) [reply superset/utils/log.py:95]: Naming is hard. I'm a fan of measure twice cut once, as renaming things (or dealing with legacy names) in the future is always somewhat painful.

Comment (etr2460) [reply superset/utils/log.py:95]: ah, i see the `path_no_param` agreement from the previous pr. tbh, CI is sooo slow tonight i probably won't merge this until tomorrow. So i'll make the change now.

**Comment by etr2460** [comment]
Since CI is green, i'm going to merge this now. Feel free to follow up with any naming PRs today or next week

**Comment by ktmud** [comment]
> Since CI is green, i'm going to merge this now. Feel free to follow up with any naming PRs today or next week

@etr2460 @mistercrunch #11927 is ready for review

**Pull request merged**
